docs/rfcs/2024-04-01-super-peers-2.md

```diff
@@ -0,0 +1,262 @@
+# Large public grups / channels
+
+This document describes specific design elements for the MVP of [the groups based on super-peers](./2024-03-14-super-peers.md).
+
+## Super-peer members
+
+There are two possible design approaches for super-peer members:
+
+1. Separate non-participating members that can only be added as a super-peers and do not have their own roles in the group, can't send their own messages or administer the group.
+
+2. Super-peer being a function of any member, irrespective of their role.
+
+While approach 1 can be simpler, it has its downsides:
+- more complex migration of the existing groups - e.g., directory service cannot become a super-peer while remaining group admin (or moderator, if we add a new role).
+- less usable clients - users can have desktop clients with good internet connectivity and sufficient computing resources to effectively host several medium size groups, even with SQLite database.
+- it makes super-peers more like servers, being both unusable as clients and also requiring larger concurrency, and therefore increasing centralization.
+
+Therefore, the approach 2 looks more attractive, when super-peers must have some role in the group, and the super peers that cannot send messages will have observer role.
```

Comment:
it seems strange to allow non-admins be super-peers, as there supposedly should be higher level of trust to them since they're relaying all messages